tests/letstest/multitester.py

```diff
@@ -402,42 +405,47 @@ def cleanup(cl_args, instances, targetlist):
 
 def main():
     # Fabric library controlled through global env parameters
-    env.key_filename = KEYFILE
-    env.shell = '/bin/bash -l -i -c'
-    env.connection_attempts = 5
-    env.timeout = 10
-    # replace default SystemExit thrown by fabric during trouble
-    class FabricException(Exception):
-        pass
-    env['abort_exception'] = FabricException
+    fab_config = Config(overrides={
+        "connect_kwargs": {
+            "key_filename": KEYFILE,
+        },
+        "run": {
+            "echo": True,
+            "pty": True,
+        },
+        "timeouts": {
+            "connect": 10,
+        },
+    })
+    local_cxn = Connection('localhost', config=fab_config)
 
     # Set up local copy of git repo
     #-------------------------------------------------------------------------------
     print("Making local dir for test repo and logs: %s"%LOGDIR)
-    local('mkdir %s'%LOGDIR)
+    local_cxn.local('mkdir %s'%LOGDIR)
 
     # figure out what git object to test and locally create it in LOGDIR
     print("Making local git repo")
     try:
         if cl_args.pull_request != '~':
             print('Testing PR %s '%cl_args.pull_request,
                   "MERGING into master" if cl_args.merge_master else "")
-            execute(local_git_PR, cl_args.repo, cl_args.pull_request, cl_args.merge_master)
+            local_git_PR(local_cxn, cl_args.repo, cl_args.pull_request, cl_args.merge_master)
         elif cl_args.branch != '~':
             print('Testing branch %s of %s'%(cl_args.branch, cl_args.repo))
-            execute(local_git_branch, cl_args.repo, cl_args.branch)
+            local_git_branch(local_cxn, cl_args.repo, cl_args.branch)
         else:
             print('Testing master of %s'%cl_args.repo)
-            execute(local_git_clone, cl_args.repo)
-    except FabricException:
+            local_git_clone(local_cxn, cl_args.repo)
+    except Exception:
```

Comment:
HA funny you ask. Here's what their documentation helpfully says: 

> The old functionality leaned too far in the “everything is a DSL” direction & didn’t offer enough value to offset how it gets in the way of experienced Pythonistas.
> 
> These functions have been removed in favor of “just raise an exception” (with one useful option being Invoke’s Exit) as exception handling feels more Pythonic than thin wrappers around sys.exit or having to except SystemExit: and hope it was a SystemExit your own code raised!

and

> Aborting as a concept is gone, just raise whatever exception seems most reasonable to surface to an end user, or use Exit. See also Utilities.

But it doesn't say what exception it actually raises internally. I'll double check in the code.

Reply:
Oh wow.

Well, taking the time to understand our code/dependencies is probably worth it, but if you want, I think it'd be OK to just catch `BaseException` here since it should catch everything.

Reply:
Looks like there's one small case where it might raise `SystemExit`: https://github.com/pyinvoke/invoke/blob/0de7020aa908d4a1afc5a7287e29196a657c292a/integration/_support/respond_both.py

Otherwise, everything inherits from `Exception` now:
https://github.com/pyinvoke/invoke/blob/f9b5d918cca93d36cb242f4a235271da4e4281f2/invoke/exceptions.py
https://github.com/fabric/fabric/blob/bdbae33842855328ad7106fab7fd8db34a8667af/fabric/exceptions.py